GitUI/CommandsDialogs/RefLogDialog/FormReflog.cs

```diff
@@ -0,0 +1,173 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text.RegularExpressions;
+using System.Windows.Forms;
+using ResourceManager;
+
+namespace GitUI.CommandsDialogs.RefLogDialog
+{
+    public partial class FormReflog : GitModuleForm
+    {
+        private readonly TranslationString _continueResetCurrentBranchText = new TranslationString("Are you sure you want to reset the current branch to this commit?");
+        private readonly TranslationString _continueResetCurrentBranchCaptionText = new TranslationString("Reset the current branch to a specific commit.");
+
+        readonly Regex _regexReflog = new Regex("^([^ ]+) ([^:]+): (.+): (.+)$", RegexOptions.Compiled);
+
+        private string _currentBranch;
+        private bool _isBranchCheckedOut;
+
+        public FormReflog(GitUICommands uiCommands)
+            : base(uiCommands)
+        {
+            InitializeComponent();
+            Translate();
+        }
+
+        private void FormReflog_Load(object sender, EventArgs e)
+        {
+            _currentBranch = UICommands.Module.GetSelectedBranch();
+            linkCurrentBranch.Text = "current branch (" + _currentBranch + ")";
+            _isBranchCheckedOut = _currentBranch != "(no branch)";
+            resetCurrentBranchOnThisCommitToolStripMenuItem.Enabled = UICommands.GitModule.IsDetachedHead();
+
+            var branches = new List<string> { "HEAD" };
+            branches.AddRange(UICommands.Module.GetRefs(false, true).Select(r => r.Name));
+            Branches.DataSource = branches;
+            DisplayRefLog();
```

Comment:
It seems that assigning the DataSource property, in the line above, triggers the Branches_SelectedIndexChanged. That causes the reflog command being executed twice at creating the dialog.